packages/dds/merge-tree/src/test/beastTest.spec.ts

```diff
@@ -1432,7 +1432,7 @@ export class DocumentTree {
 				});
 				this.pos++;
 			} else {
-				const trid = docNode.name + this.ids[docNode.name].toString();
+				const trid = docNode.name + this.ids[docNode.name]?.toString();
```

Comment:
nit: in test code that doesn't obviously fail with a more clear error, I'd much prefer we fail faster. So either this:

```suggestion
				const trid = docNode.name + this.ids[docNode.name]!.toString();
```

or the line you have, followed by

```
assert(trid !== undefined);
```

you could make this argument in some production code as well, but it's much more subjective...